cmd/juju/application/deploy_test.go

```diff
@@ -83,13 +86,64 @@ func (c fakeCharmstoreClientShim) WithChannel(channel csparams.Channel) charmsto
 	return fakeCharmstoreClientShim{*cstore}
 }
 
+// testcharmsCharmstoreClientShim allows a jjcharmstore.ChannelAwareFakeClient to
+// masquarade as a testcharms.CharmstoreClient implementer
+type testcharmsCharmstoreClientShim struct {
```

Comment:
charmstoreClientShim

I don't think we need both shims?
They both embed a jjcharmstore.ChannelAwareFakeClient
We just just have the one that has all the methods. Make naming less convoluted also.

Reply:
> "Make the naming less convoluted" 

appy to change to whatever you would like. I went around in circles for a while, but ended up here as the least convoluted thing I could think of that's also not otherwise taken.